File: 2012/submissions/Wesley Taylor/1/haskell/problem_1.hs
import System.IO (hFlush, stdout)
import Data.List (findIndex)
import Data.Maybe (isJust, fromJust)

getWord :: IO String
getWord = do
        putStr "Enter a word: "
        hFlush stdout
        getLine

getSentence :: IO [String]
getSentence = do
         putStr "Enter a sentence: "
         hFlush stdout
         words `fmap` getLine

wordIndex :: String -> [String] -> Maybe Int
wordIndex word sentence = findIndex (== word) sentence

wordFoundStr :: Maybe Int -> String
wordFoundStr index = "Word was found at index " ++
                     (show (fromJust index))    ++
                     "."

printResult :: Maybe Int -> IO ()
printResult index = if isJust index
                       then putStrLn $ (wordFoundStr index)
                       else putStrLn "Word was not found."

main = do
     wordToFind   <- getWord
     sentence     <- getSentence
     let index    =  wordIndex wordToFind sentence
     printResult index

File: 2012/submissions/Wesley Taylor/1/perl/problem_1.pl
#!/bin/env perl

use strict;
use warnings;
use List::MoreUtils qw(firstidx);

print "Please enter a word: ";
my $word_to_find = <>;
$word_to_find =~ s/^\s+|\s+$//g;

print "Sentence: ";
my $sentence = <>;
my @words = split( /\s+/, $sentence );
my $index = firstidx { /\b$word_to_find\b/ } @words;
if ($index == -1)
{
    print "Word not found.\n"
}
else
{
    print "Word found at index $index.\n";
}
